### PR TITLE
[CL] Improve data structure of Phreeqc Interface

### PR DESCRIPTION
**Pull request by renchao-lu**
- Use base class `ostream` for output; 
- Operator overloading on class templates;

### CODE REVIEW

File: ChemistryLib/PhreeqcIO.cpp [hunk 1]
```diff
@@ -19,6 +19,18 @@
 
 namespace ChemistryLib
 {
+namespace
+{
+template <typename DataBlock>
+std::ostream& operator<<(std::ostream& os,
+                         std::vector<DataBlock> const& data_blocks)
+{
+    std::copy(data_blocks.begin(), data_blocks.end(),
```
Comment (endJunction) [inline ChemistryLib/PhreeqcIO.cpp]: I'd think, it could be
```c++
return std::copy(...);
```

File: ChemistryLib/Output.cpp [hunk 2]
```diff
@@ -7,69 +7,69 @@
  *
  */
 
-#include <fstream>
+#include <iostream>
```
Comment (endJunction) [inline ChemistryLib/Output.cpp]:
```suggestion
#include <ostream>
```

Comment (renchao-lu) [reply ChemistryLib/Output.cpp]: Changed as suggested except in the `PhreeqcIO.cpp` where operator `>>` (istream) is used.

File: ChemistryLib/PhreeqcIO.cpp (hunk 1 above)
Comment (renchao-lu) [reply ChemistryLib/PhreeqcIO.cpp]: May discuss tomorrow...

**Pull request merged**